### DISCUSSION TITLE
Hide paginator if table empty

### DISCUSSION THREAD

**Discussion by rychagov** [primefaces/community#2548] [PrimeNG]
Hi! 
**What we have:** Primeng table with state (local/session) and paginator.
**What we would like to have:** hide paginator when there is no data in table.

We cannot use `[paginator]="someVariableOrFunctionToCheckData`" because restoring paginator settings from storage depends on it.
And we have no data on page load. So, when we init component paginator="false" and we cannot restore data.
Any ideas how to reach functionality?

**Comment by ibrahimkelly** [comment]
> Hi! **What we have:** Primeng table with state (local/session) and paginator. **What we would like to have:** hide paginator when there is no data in table.
> 
> We cannot use `[paginator]="someVariableOrFunctionToCheckData`" because restoring paginator settings from storage depends on it. And we have no data on page load. So, when we init component paginator="false" and we cannot restore data. Any ideas how to reach functionality?

`[paginator]="values.length > 0 ? true : false"`

**Reply by rychagov** [reply]
Nope, because we have state in localStorage with paginator options.
So, the order is:
1. Loading page
2. Init table component (with empty data)
3. Request data to place in the table

The problem of `[paginator]="values.length > 0"` is checking **paginator** option on component **init** to get **state** from **localStorage**.
So, if we have no data on component init **paginator** will be **false**, so, when we get some data for table we cannot show options from **storage**.
